pkg/kubelet/kubelet_pods_test.go

```diff
@@ -195,6 +195,50 @@ func writeHostsFile(filename string, cfg string) (string, error) {
 	return tmpdir, os.WriteFile(filepath.Join(tmpdir, filename), []byte(cfg), 0644)
 }
 
+func TestEnsureHostsFile(t *testing.T) {
+	testCases := []struct {
+		hostIPs        []string
+		hostName       string
+		hostDomainName string
+		hostAliases    []v1.HostAlias
+		useHostNetwork bool
+	}{
+		{
+			hostIPs:        []string{"1.1.1.1"},
+			hostName:       "podFoo",
+			hostDomainName: "domainFoo",
+			hostAliases:    []v1.HostAlias{{IP: "123.45.67.89", Hostnames: []string{"foo", "bar", "baz"}}},
+			useHostNetwork: false,
+		},
+		{
+			hostIPs:        []string{"1.1.1.1"},
+			hostName:       "podFoo",
+			hostDomainName: "domainFoo",
+			hostAliases:    []v1.HostAlias{{IP: "123.45.67.89", Hostnames: []string{"foo", "bar", "baz"}}},
+			useHostNetwork: true,
+		},
+	}
+	for _, testCase := range testCases {
```

Comment:
It would be better to add description for each test case and run them with with t.Run, for example [like here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_pods_test.go#L1679)